numpy/core/include/numpy/npy_cpu.h

```diff
@@ -115,4 +116,110 @@
+    NPY_CPU_FEATURE_XOP               = 12,
+    NPY_CPU_FEATURE_FMA4              = 13,
+    NPY_CPU_FEATURE_AVX2              = 14,
+
+    NPY_CPU_FEATURE_AVX512F           = 30,
+    NPY_CPU_FEATURE_AVX512CD          = 31,
+    NPY_CPU_FEATURE_AVX512ER          = 32,
+    NPY_CPU_FEATURE_AVX512PF          = 33,
+    NPY_CPU_FEATURE_AVX5124FMAPS      = 34,
+    NPY_CPU_FEATURE_AVX5124VNNIW      = 35,
+    NPY_CPU_FEATURE_AVX512VPOPCNTDQ   = 36,
+    NPY_CPU_FEATURE_AVX512BW          = 37,
+    NPY_CPU_FEATURE_AVX512DQ          = 38,
+    NPY_CPU_FEATURE_AVX512VL          = 39,
+    NPY_CPU_FEATURE_AVX512IFMA        = 40,
+    NPY_CPU_FEATURE_AVX512VBMI        = 41,
+    NPY_CPU_FEATURE_AVX512VNNI        = 42,
+    NPY_CPU_FEATURE_AVX512VBMI2       = 43,
+    NPY_CPU_FEATURE_AVX512BITALG      = 44,
+
+    /* X86 CPU Groups */
+    /*(F,CD)*/
+    NPY_CPU_FEATURE_AVX512_COMMON     = 100,
+    /* Knights Landing (F,CD,ER,PF)*/
+    NPY_CPU_FEATURE_AVX512_KNL        = 101,
+    /* Knights Mill    (F,CD,ER,PF,4FMAPS,4VNNIW,VPOPCNTDQ) */
+    NPY_CPU_FEATURE_AVX512_KNM        = 102,
+    /* Skylake-X       (F,CD,BW,DQ,VL) */
+    NPY_CPU_FEATURE_AVX512_SKX        = 103,
+    /* Cannon Lake     (F,CD,BW,DQ,VL,IFMA,VBMI) */
+    NPY_CPU_FEATURE_AVX512_CNL        = 104,
+    /* Cascade Lake    (F,CD,BW,DQ,VL,IFMA,VBMI,VNNI) */
+    NPY_CPU_FEATURE_AVX512_CEL        = 105,
+    /* Ice Lake        (F,CD,BW,DQ,VL,IFMA,VBMI,VNNI,VBMI2,BITALG,VPOPCNTDQ) */
+    NPY_CPU_FEATURE_AVX512_ICL        = 106,
+
+    /* IBM/POWER VSX */
+    /* POWER7 */
+    NPY_CPU_FEATURE_VSX               = 200,
+    /* POWER8 */
+    NPY_CPU_FEATURE_VSX2              = 201,
+    /* POWER9 */
+    NPY_CPU_FEATURE_VSX3              = 202,
+
+    /* ARM */
+    NPY_CPU_FEATURE_NEON              = 300,
+    NPY_CPU_FEATURE_NEON_FP16         = 301, /* VFPV3_F16 */
+    NPY_CPU_FEATURE_NEON_VFPV4        = 302, /* FMA */
+    NPY_CPU_FEATURE_ASIMD             = 303, /* Advanced SIMD */
+    NPY_CPU_FEATURE_FPHP              = 304, /* ARMv8.2 half-precision */
+    NPY_CPU_FEATURE_ASIMDHP           = 305, /* ARMv8.2 half-precision vector arithm */
+    NPY_CPU_FEATURE_ASIMDDP           = 306, /* ARMv8.2 dot product */
+    NPY_CPU_FEATURE_ASIMDFHM          = 307, /* ARMv8.2 single&half-precision multiply */
+
+    NPY_CPU_FEATURE_MAX
+};
+
+/*
+ * Initialize CPU features
+ *
+ * This function
+ *  - detect CPU features avalibilty in runtime
+ *  - check the sanity of CPU baseline features
+ *  - disable CPU dispatched features in runtime,
+ *    depend on given values of environment variable 'NPY_DISABLE_CPU_FEATURES'
+ *
+ * It also trigger Py runtime error when
+ *  - NumPy build has CPU baseline features
+ *    that aren't supported by the current hardware
+ *  - The misuse of environment variable 'NPY_DISABLE_CPU_FEATURES'
+ *
+ * return 0 on success otherwise return -1
+*/
+int npy_cpu_init(void);
+
+/*
+ * return 0 if CPU feature isn't available
```

Comment:
what does it return on success?

Reply:
It should return 1 if the compiler respects standard C anyway, returned data should treat as boolean.